### PR TITLE
Tickets/dm 41371

### PR DESCRIPTION
**Pull request by teweitsai**
* Support the interrupt of actuator bump test and publish the status of bump test.

### CODE REVIEW

File: python/lsst/ts/m2/csc.py [hunk 1]
```diff
@@ -624,6 +640,16 @@ async def do_disable(self, data: salobj.BaseMsgType) -> None:
...
 
+    def _is_bump_test_done(self) -> bool:
+        """The bump test is done or not.
+
+        Returns
+        -------
+        `bool`
+            True if the bump test is done. Otherwise, False.
+        """
+        return self._task_bump_test.done() or self._task_bump_test.cancelled()
```
Comment (tribeiro) [inline python/lsst/ts/m2/csc.py:651]: A "cancelled" task is also considered "done", so you can do just `return self._task_bump_test.done()`

File: python/lsst/ts/m2/csc.py [hunk 2]
```diff
@@ -1151,6 +1212,41 @@ async def _bump_actuator(
...
+        ----------
+        data : `object`
+            Data of the SAL message.
+        """
+
+        await self.cmd_killActuatorBumpTest.ack_in_progress(
+            data, timeout=self.COMMAND_TIMEOUT
+        )
+
+        self._task_bump_test.cancel()
```
Comment (tribeiro) [inline python/lsst/ts/m2/csc.py:1248]: In an excess of caution, I recommend waiting for the task to finish before returning. 

I'd also trow in an info log message before cancelling it.

Something like:

```
self.log.info("Killing bump test.")
self._task_bump_test.cancel()
try:
    await self._task_bump_test
except asyncio.CancelledError:
    self.log.debug("Bump test task cancelled.")
```

File: python/lsst/ts/m2/csc.py [hunk 3]
```diff
@@ -1151,6 +1212,41 @@ async def _bump_actuator(
...
+
+    async def _do_killActuatorBumpTest(self, data: salobj.BaseMsgType) -> None:
+        """Kill the running actuator bump test in the closed-loop control.
+
+        Parameters
+        ----------
+        data : `object`
+            Data of the SAL message.
+        """
+
```
Comment (tribeiro) [inline python/lsst/ts/m2/csc.py:1243]: I think you need to add an `self.assert_enabled()` here.

also, consider making it an immediate return (no `IN_PREOGRESS`) if the task is already cancelled;

```
if self._task_bump_test.done():
    return
```

**Review by tribeiro — APPROVED** [review]
Looks really good! I left a few inline comments for you to consider. Please, take a look before merging.

File: python/lsst/ts/m2/csc.py (hunk 1 above)
Comment (teweitsai) [reply python/lsst/ts/m2/csc.py:651]: Done in the commit: db3ca38. Thanks!

File: python/lsst/ts/m2/csc.py (hunk 2 above)
Comment (teweitsai) [reply python/lsst/ts/m2/csc.py:1248]: Done in the commit: https://github.com/lsst-ts/ts_m2/commit/db3ca385e39a6b5322b55a523e32a1b4d19a539f. Thanks!

File: python/lsst/ts/m2/csc.py (hunk 3 above)
Comment (teweitsai) [reply python/lsst/ts/m2/csc.py:1243]: Done in the commit: https://github.com/lsst-ts/ts_m2/commit/db3ca385e39a6b5322b55a523e32a1b4d19a539f. Thanks!

**Pull request merged**